src/app/features/create/create.component.spec.ts

```diff
@@ -0,0 +1,215 @@
+        return;
+      });
+      const setErrorsSpy = jest.spyOn(errorService, 'setErrors').mockImplementation();
+      component.isFormValid;
+      expect(setErrorsSpy).toHaveBeenCalledTimes(1);
+      expect(setErrorsSpy).toHaveBeenCalledWith([]);
+    });
+
+    it('should return vehicleForm.valid', () => {
+      const formValid = component.isFormValid;
+      expect(formValid).toBeFalsy();
+    });
+  });
+
+  describe('navigateBack', () => {
+    it('should clear all errors', () => {
+      jest.spyOn(router, 'navigate').mockImplementation();
+
+      const clearErrorsSpy = jest.spyOn(errorService, 'clearErrors');
+
+      component.navigateBack();
+
+      expect(clearErrorsSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should navigate back to the previous page', () => {
+      const navigateSpy = jest.spyOn(router, 'navigate').mockImplementation(() => Promise.resolve(true));
+
+      component.navigateBack();
+
+      expect(navigateSpy).toBeCalledWith(['..'], { relativeTo: route });
+    });
+  });
+
+  describe('handleSubmit', () => {
+    it('should do nothing if the form is not valid', () => {
+      const formUniqueSpy = jest.spyOn(component, 'isFormValueUnique').mockImplementation();
+      component.handleSubmit();
+      expect(formUniqueSpy).toHaveBeenCalledTimes(0);
+    });
+
+    it('should do nothing if the form value not unique', () => {
+      const isFormValid = jest.spyOn(component, 'isFormValid', 'get').mockReturnValue(true);
+      const updateEditingSpy = jest.spyOn(techRecordService, 'updateEditingTechRecord');
+      const navigateSpy = jest.spyOn(router, 'navigate');
+      const generateTechREcordSpy = jest.spyOn(techRecordService, 'generateEditingVehicleTechnicalRecordFromVehicleType');
+      component.handleSubmit();
+
+      expect(isFormValid).toReturn();
+      expect(updateEditingSpy).toHaveBeenCalledTimes(0);
+      expect(generateTechREcordSpy).toHaveBeenCalledTimes(0);
+      expect(navigateSpy).toHaveBeenCalledTimes(0);
+    });
+
+    it('should navigate to hydrate when successful', fakeAsync(() => {
+      jest.spyOn(component, 'isFormValid', 'get').mockReturnValue(true);
+      jest.spyOn(component, 'isFormValueUnique').mockImplementation(() => Promise.resolve(true));
+      const routerSpy = jest.spyOn(router, 'navigate').mockImplementation(() => Promise.resolve(true));
+      const updateEditingSpy = jest.spyOn(techRecordService, 'updateEditingTechRecord');
+
+      component.vehicle = { techRecord: [{ vehicleType: VehicleTypes.HGV } as TechRecordModel] };
+      component.handleSubmit();
+      tick();
+
+      expect(routerSpy).toHaveBeenCalledWith(['../create/new-record-details'], { relativeTo: route });
+    }));
+  });
+
+  describe('isVinUnique', () => {
+    it('should call isUnique with an emptry string and the type of vin', async () => {
+      const isUniqueSpy = jest.spyOn(techRecordService, 'isUnique').mockImplementation(() => of(true));
```

Comment:
```suggestion
      const isUniqueSpy = jest.spyOn(techRecordService, 'isUnique').mockResolvedValue(true);
```
Just because the `isUnique` method returns a Promise, not an observable

Reply:
And same for the other tests in here

Reply:
nope

Reply:
Ignore me of course 😆